lib/spack/spack/package.py

```diff
@@ -1217,6 +1236,15 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class PythonExtension(Package):
+    def setup_dependent_environment(self, env, dependent_spec):
+        pass
+
+    def setup_environment(self, env):
+        site_packages = glob.glob(join_path(self.spec.prefix.lib, "python*/site-packages"))
+        if site_packages:
+            env.prepend_path('PYTHONPATH', site_packages[0])
```

Comment:
I think you don't actually need this.

The `setup_dependent_environment` of extendees is called automatically for extensions [here](https://github.com/LLNL/spack/pull/552/files#diff-c8f14b8e4ac7661fb9ca8ecc00bc2ebcR286), so the `site-packages` and `PYTHONPATH` directories will already be set up automatically before `install()`.  This is actually necessary for `setuptools` to do the right thing.

Probably what should happen is that `setup_dependent_environment` should use the `env` object to set `PYTHONPATH` -- then it will be included in the module and in the `install()` environment.  Extensions can obviously implement their own `setup_environment` if needed, but this way they don't have to do so just for the basic `PYTHONPATH`.

The other thing about the code above is that it makes the extension writer have to remember two things instead of one:

  1. add `extends('python'), and
  2. remember to derive from `PythonExtension`

I think only the first should be necessary.  I could see users getting frustrated if they do 1 and not 2.  Am I missing something?